.github/workflows/build-macos.yml

```diff
@@ -55,26 +56,6 @@ jobs:
             ~/.triton/nvidia
             ~/.triton/json
           key: ${{ runner.os }}-${{ runner.arch }}-llvm-${{ steps.cache-key.outputs.llvm }}-nvidia-${{ steps.cache-key.outputs.nvidia }}-json-${{ steps.cache-key.outputs.json }}
-      - # Cache ~/.cache/ccache to speed up compilation.
-        #
-        # On branch `main` we always start from an empty cache, i.e. we skip the
-        # "restore" step.  This is to prevent the caches from accumulating stale
-        # files over time.
-        name: Restore cache of ccache and Triton compilation artifacts
-        id: restore-build-cache
-        if: github.ref != 'refs/heads/main'
-        uses: actions/cache/restore@v4
```

Comment:
I don't see this action being used in amd or nvidia. So just taking it out

Reply:
@ThomasRaoux why did you remove the caching in #7076?

Reply:
> @ThomasRaoux why did you remove the caching in #7076?

It seemed like it was causing the llvm cache to be invalidated/missed more often which caused dramatic slow down. I couldn't fully explain why though so maybe I had missed something

Reply:
One thing worth noting is that the github actions cache has a storage limit of 10GB and the nvidia-llvm-json combined files are over 1GB each.
See: https://github.com/triton-lang/triton/actions/caches

Perhaps we could pay for more storage, or use another method to store some of the caches?